app/models/katello/content_view_puppet_module.rb

```diff
@@ -8,6 +8,7 @@ class ContentViewPuppetModule < Katello::Model
     validates :content_view_id, :presence => true
     validates :name, :uniqueness => { :scope => :content_view_id }, :allow_blank => true
     validates :uuid, :uniqueness => { :scope => :content_view_id }, :allow_blank => true
+    validate :uuid_associated_with_puppet_module
```

Comment:
This is not a join table, and does not necessarily represent a `PuppetModule` which is where the confusion probably is. Trust me, this got me the first few times as well. The idea behind this class is largely captured at [1] wherein a user can add a puppet module by "name" and "author" choosing to have the "latest" puppet module used. So, instead of locking it down to a particular version (and thus a specific puppet module), the user can opt to have the latest version of the puppet module pulled in at publish time every time.


[1] https://github.com/Katello/katello/pull/6093/files#diff-b4b4aa0faa5bb64c59dd46b0a744620cR30